docs/docs/features/conditional-layers.md

```diff
@@ -0,0 +1,51 @@
+---
+title: Conditional Layers
+---
+
+Conditional layers support activates a particular layer (called the then-layer) when all layers in a
+specified set (called the if-layers) are active. This feature generalizes what's commonly known as
+tri-layer support, allowing activation of two layers (usually called "lower" and "raise") to trigger
+a third (usually called "adjust").
+
+Another way to think of this feature is as a simple combo system for layers, just like the usual
+[combos for behaviors](combos.md).
+
+## Configuration
+
+Conditional layers are configured via a `conditional_layers` node in your `.keymap` file as follows:
+
+```
+/ {
+    conditional_layers {
+        compatible = "zmk,conditional-layers";
+        tri_layer {
+            if-layers = <1 2>;
+            then-layer = <3>;
+        };
+    };
+};
+```
+
+Each conditional layer configuration may have whatever name you like, but it's helpful to chose
+something self explanatory like `tri_layer`. The following properties are supported:
+
+- The `if-layers` property specifies a set of layer numbers, all of which must be active for the
+  conditional layer to trigger.
+- The `then-layer` property specifies a layer number that should be activated if and only if all the
+  layers specified in the `if-layers` property are active.
+
+Therefore, in this example, layer 3 ("adjust") will be activated if and only if both of layers 1
+("lower") and 2 ("raise") are active.
+
+:::info
+One conditional layer configuration can trigger another, possibly many times in succession; however,
+this process must eventually end since there are only a finite number of layers.
```

Comment:
I had some trouble parsing this statement. Does this mean that an infinite loop isn't technically possible? I assume so, if `if-layers` is constrained to have at least two elements.

Reply:
Yeah, as far as a I know, an infinite loop shouldn't be possible since the implementation only tries to activate a layer if it's not already active, and so in the worst case you end up done when every layer is active. But honestly, on further thought I don't think that detail is really important, so I pruned down the note text. WDYT?

Reply:
I think that's good, the user doesn't really need to know whether it can cause an issue anyway, especially if it cannot :)